actionview/lib/action_view/template.rb

```diff
@@ -168,6 +168,10 @@ def translate_location(backtrace_location, spot)
       if handler.respond_to?(:translate_location)
         handler.translate_location(spot, backtrace_location, encode!)
       else
+        ActionView.deprecator.warn(<<-MSG.squish)
+          The templating engine being used does not define `translate_location`
+          so Rails is unable to highlight the source location inside the template.
```

Comment:
After discussing with @k0kubun we're removing the deprecation in #47124 because it is non-trivial to implement in HAML and also probably Slim. When/if template engines implement this they will get error highlighting, without it the error will be displayed the same as before we had this information.